### PR TITLE
Add yarn

### LINKED ISSUE #181: Switch to yarn
Both for shep and the generated project structure.

### PR DESCRIPTION
**Pull request by southpolesteve**
Closes #181

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -256,8 +256,8 @@ It was called 'shepherd' at first because it was helpful for dealing with *lamb*
...
 Pull requests welcome!
 
-Test: `npm test`
+Test: `yarn test`
 
-Rebuild on file change: `npm run compile -- -w`
+Rebuild on file change: `yarn compile -- -w`
 
-Publish: `npm run pub` "publish" is reserved by npm
+Publish: `yarn pub` "publish" is reserved by npm
```
Comment (zfoster) [inline README.md]: is publish reserved by `yarn`?

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
 language: node_js
 node_js:
   - stable
-install:
-  - npm install
+
+cache:
+  yarn: true
+  directories:
```
Comment (gdub22) [inline .travis.yml]: Should only need `cache: yarn`

File: README.md [hunk 3]
```diff
@@ -22,8 +22,8 @@ Shep will require your amazon credentials and will load them using the same meth
 ### Installation
 
 ```bash
-npm install -g shep
-npm install --save-dev shep // the global shep will run the local shep
+yarn install -g shep
```
Comment (gdub22) [inline README.md]: `yarn global add shep`

File: README.md [hunk 4]
```diff
@@ -22,8 +22,8 @@ Shep will require your amazon credentials and will load them using the same meth
...
+yarn install --save-dev shep // the global shep will run the local shep
```
Comment (gdub22) [inline README.md]: `yarn add shep --dev`

File: README.md (hunk 1 above)
Comment (southpolesteve) [inline README.md]: yes. You can't overwrite it. fwiw `yarn publish` is better than `npm publish`. It has a np-lite kind of workflow

**Pull request merged**